### PR TITLE
Fixed C deprecated enum support. Added C++03 support for examples

### PR DESCRIPTION
**Pull request by ethouris**

### CODE REVIEW

File: srtcore/srt.h [hunk 1]
```diff
@@ -185,6 +192,29 @@ typedef enum SRT_SOCKOPT {
...
+
+    // In C, declare them as a separate enum, as it's not an error in C to mix enum types
+    typedef enum SRT_SOCKOPT_DEPRECATED
+    {
+    #define SRT_DECLARE_DEPRECATED_OPT(name, value) \
+        name SRT_ATR_DEPRECATED = value,
+
+#else
+#define SRT_NO_DEPRECATED_ENUM
+#define SRT_DECLARE_DEPRECATED_OPT(name, value) \
```
Comment (rndi) [inline srtcore/srt.h]: This does not look right: no deprecated values will be defined. Also I think we still want to have `enum SRT_SOCKOPT_DEPRECATED {}` for C compatibility.

Comment (ethouris) [reply srtcore/srt.h]: Ok, I've rewritten it completely. Tested with an old compiler in C, C++ and C++11 modes. Reports the misused deprecated option perfectly in all cases.

Comment (rndi) [reply srtcore/srt.h]: With gcc 4.4.1 deprecated option is not reported. Instead, I get  
> warning: 'SRT_SOCKOPT_DEPRECATED' is deprecated

Comment (ethouris) [reply srtcore/srt.h]: Intended is that you get a warning when this option is **used**, no matter what the warning says. If it works this way, that's good enough. I checked 4.4.1 and it reports exactly the line where the deprecated enum is used - even though the message isn't too readable.

**Pull request merged**